sources/HUBComponentModelImplementation.m

```diff
@@ -190,6 +190,31 @@ - (NSString *)debugDescription
 
 #pragma mark - HUBComponentModel
 
+- (NSIndexPath *)indexPath
+{
+    // Since we're traversing from this model through the parents, we will record the indices in reverse order.
+    // Capture them in a mutable array for easy reversal.
+    NSMutableArray<NSNumber *> *reversedIndexPath = [NSMutableArray array];
```

Comment:
Since this is not an index path, shouldn't we name it something like `reversedIndexes`? (just a thought)